### ISSUE TITLE
Support for windows

### ISSUE THREAD

**Issue by MStoykov** [codecov/codecov-action#7]
Given that github can run windows (and macOs) builds on top of linux, it will be nice if the official codecov action can run on all three. 

I see that there is [codecov-exe](https://github.com/codecov/codecov-exe) , so I suppose this is being worked on ?

**Comment by adam7** [comment]
Same for macOS so I don't think codecov.exe will resolve the issue.

Build failure is because this is a Docker Action which are only supported on Linux. Porting to a JavaScript Action would resolve the issue because they are supported on Windows, macOS and Linux. 

https://help.github.com/en/articles/about-actions#types-of-actions

**Comment by ibrahim0814** [comment]
Hi all,

Thank you for bringing this up. I've updated our README to reflect this particular caveat. GitHub currently allows us to build two types of Actions - Docker and JS. We decided to use a Docker action since it allows us to wrap our bash uploader because our bash uploader is the only uploader the Codecov staff actively maintains. Of course, this comes at the cost of sacrificing Windows and macOS build capabilities.

Perhaps in the future, we will support these other operating systems, but for the time being, I do apologize that this action's functionality is limited to only Linux

Ib
